### DISCUSSION TITLE
Listening to multiple nodes

### DISCUSSION THREAD

**Discussion by george-kar** [SchneiderInfosystems/FB4D#153] [Q&A] [answered]
In Firebase RealtimeDB web sdk you are able to install listener for multiple nodes like

- Node1
+ Node2
--- SubNode 1
--- SubNode 2
+ Node3
--- SubNode 3

You can add one listener to Node1, SubNode2 and SubNode3 with one call/socket.

Now for Firebase Delphi implementation, as I see on your documentation https://github.com/SchneiderInfosystems/FB4D/wiki/FB4D-Reference-IRealTimeDB#install-a-listener-for-changes-in-a-json-node

I can add a listener to a particular JSON node, so in my case I have a) to add 3 listeners to listen those 3 specific nodes or b) add a listener to the whole tree. I dont like either because a) I have to open 3 sockets or b) reading the whole tree will bring me unnecessary changes/data from other nodes.

Any thoughts about it?

**Comment by SchneiderInfosystems** [comment]
In this situation, I reconmend to use more than one instance of IRealTimeDB. Please inform me here, if this solves your task.

**Reply by george-kar** [reply]
that will open a new socket for each node? or use one for everything as web sdk?

**Comment by george-kar** [comment, marked as answer]
I was able to listen two different nodes but using two IFirebaseEvent and one IRealTimeDB, instead of two IRealTimeDB instances as you proposed. I think its much better. Example code for other users:

```
fAuth := TFirebaseAuthentication.Create('apikey');
  fRealTimeDB := TRealTimeDB.CreateByURL('https://...', fAuth);

  fFirebaseEvent := fRealTimeDB.ListenForValueEvents(
    SplitString(edtRTDBEventPath.Replace('\', '/'), '/'),
    OnRecData, OnRecDataStop, OnRecDataError);

  fFirebaseEvent2 := fRealTimeDB.ListenForValueEvents(
    SplitString(edtRTDBEventPath2.Replace('\', '/'), '/'),
    OnRecData, OnRecDataStop, OnRecDataError);
```

**Reply by SchneiderInfosystems** [reply]
I agree that this is the right way to solve this request. Using two instances of IRealTimeDB is only necessary if you need to access via different threads without any thread synchronization.
